acme/acme/messages.py

```diff
@@ -92,10 +93,10 @@ def code(self):
             return code
 
     def __str__(self):
-        return ' :: '.join(
-            part for part in
+        return b' :: '.join(
+            part.encode('ascii', 'backslashreplace') for part in
```

Comment:
Something I forgot to make note of is that when the error in #4273 occurs and this code gets executed, in either Python 2 or 3 (I can't remember which), when the output describing the error/non-ascii domain is printed, it prints literal bytes (e.g. \xBlah\Blah) instead of the domain name. I couldn't find a quick way to print the unicode characters nicely everywhere :(

This is because of this line. I had the choice of using `backslashreplace` or `ignore`, and literal bytes seemed better for debugging purposes than just throwing out anything that couldn't be encoded into ascii ...